### PR TITLE
trigger always a custom even on axa change fi the WC has a value prop…

### LINKED ISSUE #376: Trigger custom events of every component
## Are you reporting a bug or a feature?

 - [x] Feature

## Expected Behavior

I expect that every custom element, triggers any internal events necessary for the outer world as a custom event, which can be `defaultPrevented`, `propagationStopped`, etc.

And I expect all available events for consumers to be properly documented (with [jsdoc `@event`](http://usejsdoc.org/tags-event.html)).

All event names shall share a consistent naming, like `axa-click`. **Important** those names have to be simple, no fancy separators shall be used for integration purposes.

**Related:**
 - Integration POC #330 

**Affected Components**
- `<axa-heckbox>`
- `<axa-choice>`
- `<axa-forminput>`
- `<axa-radio>`
- `<axa-accordion>`
- `<axa-accordion-item>`
- `<axa-button>`
- `<axa-dropdown>`
- `<axa-footer-languages>`
- `<axa-footer-legals>`
- `<axa-footer-social>`
- `<axa-header-burger>`
- `<axa-header-languages>`
- `<axa-header-logo>`
- `<axa-header-mobile-languages>`
- `<axa-header-mobile-navigation>`
- `<axa-header-mobile-others>`
- `<axa-header-navigation>`
- `<axa-header-others>`
- `<axa-header-search>`
- `<axa-header-sub-navigation>`
- `<axa-link>`

## Actual Behavior

Not a single component implements any custom event for consumers.

### PR DESCRIPTION
**Pull request by LucaMele**
Fixes #376 and minor issues

### CODE REVIEW

File: src/js/abstract/hocs/with-update.js [hunk 1]
```diff
@@ -144,6 +146,11 @@ const withUpdate = Base =>
       } else { // delete attribute
         this[key] = null;
       }
+
+      // if value is updated, we presume that an axa on change event have to be triggered
+      if (name === 'value' && newValue !== null && oldValue !== newValue) {
```
Comment (AndyOGo) [inline src/js/abstract/hocs/with-update.js]: Check for updating is done above by `shouldUpdateCallback`.

Hmh do we really want to fire this for all value attributes?

File: src/demos/todomvc/todo-item.jsx [hunk 2]
```diff
@@ -119,6 +121,15 @@ class TodoItem extends Component {
     return (
       <li className="m-todo__item">
         <div className="m-todo__wrap">
+          <AXADropDownReact
+            title="Switch status"
+            value={completed ? 'selected' : 'unselected'}
```
Comment (AndyOGo) [inline src/demos/todomvc/todo-item.jsx]: I appreciate the `value` prop, I think we should follow react's principle here for all input components

Comment (LucaMele) [reply src/demos/todomvc/todo-item.jsx]: which is?

Comment (LucaMele) [reply src/demos/todomvc/todo-item.jsx]: i emulated all input principles.. all user-inputs HTML tags have value as attribute to set value, and this is what i emulated. What do you mean with "we should follow react's principle"? The sentence as it is unfortunately says nothing

Comment (AndyOGo) [reply src/demos/todomvc/todo-item.jsx]: Sorry I meant to reference all [React's Form concepts](https://reactjs.org/docs/forms.html):
- [Controlled Components](https://reactjs.org/docs/forms.html#controlled-components)
- `[value=""]` attribute over HTML spec for [`<textarea>`](https://reactjs.org/docs/forms.html#the-textarea-tag), [`<select>`](https://reactjs.org/docs/forms.html#the-select-tag)

Comment (LucaMele) [reply src/demos/todomvc/todo-item.jsx]: Ahh ok.. well i think is quite similar how we started now isn't it?

Comment (LucaMele) [reply src/demos/todomvc/todo-item.jsx]: but anyway as best practice for future design desicions, we should keep to HTML standard/way to do and not react

**Pull request merged**